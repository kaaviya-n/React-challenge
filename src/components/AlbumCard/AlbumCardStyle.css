.title {
    font-size: 15px;
    color: #ffffff;
    font-weight: 700;
    display: -webkit-box;
    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
}
.artistName {
    font-size: 14px;
    color: #b3b3b3;
}
.artistName:hover {
    border-bottom: 1px solid #b3b3b3;
}
.playlistWrapper {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
}
@media screen and (max-width: 740px) {
    .playlistWrapper {
        justify-content: center;
    }
}
.playlistCardWrapper {
    text-align: center;
    padding: 14px 16px;
    background: #181818;
    border-radius: 4px;
    margin: 6px 8px;
    transition: 0.3s;
}
.playlistCardWrapper:hover {
    background-color: #282828;
}
.albumImage {
    object-fit: cover;
    object-position: center center;
    border-radius: 8px;
    margin: 0 auto;
}