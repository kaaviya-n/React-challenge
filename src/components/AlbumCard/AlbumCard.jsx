import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import './AlbumCardStyle.css';

const AlbumCard = ({ album, onCardClick, from }) => {
	const { images, name, artists } = album;
	return (
		<div className="playlistCardWrapper" onClick={onCardClick}>
			<img
				src={images[1].url}
				alt="album"
				width="150px"
				className="albumImage"
			/>
			<div style={{ paddingTop: '10px', width: '190px' }}>
				<div className="title">{name}</div>
				<div style={{ color: '#b3b3b3', paddingTop: '7px', height: '40px' }}>
					{artists.map((artist, index) => (
						<div key={artist.id}>
							<Link
								onClick={e => e.stopPropagation()}
								to={from !== 'artist' && `/artist/${artist.id}/${artist.name}`}
								className="artistName"
							>
								{artist.name}
							</Link>
							{index !== artists.length - 1 && ', '}
						</div>
					))}
				</div>
			</div>
		</div>
	);
};

AlbumCard.propTypes = {
	album: PropTypes.object,
	onCardClick: PropTypes.func,
	from: PropTypes.string
};

export default AlbumCard;