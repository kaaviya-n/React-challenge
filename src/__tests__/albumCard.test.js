import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AlbumCard from '../components/AlbumCard/AlbumCard';

afterEach(cleanup);

const albumData = {
	'artists': [
		{
			'id': '3kHVioJpVxlazAAKQ64pC1',
			'name': 'Yuna',
		}
	],
	'images': [
		{
			'height': 640,
			'url': 'https://i.scdn.co/image/ab67616d0000b273b1494c5d42668590efcda7a2',
			'width': 640
		},
		{
			'height': 300,
			'url': 'https://i.scdn.co/image/ab67616d00001e02b1494c5d42668590efcda7a2',
			'width': 300
		},
	],
	'name': 'Y3'
};

const AlbumJSX = <AlbumCard album={albumData} />;

it('UI testing - Album card', () => {
	const tree = renderer.create(<BrowserRouter>{AlbumJSX}</BrowserRouter>).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders Album card component', () => {
	render(AlbumJSX, {wrapper: BrowserRouter});
});

it('Should render without throwing an error', () => {
	expect(render(
		AlbumJSX, {wrapper: BrowserRouter}
	)).not.toBeNull();
});
