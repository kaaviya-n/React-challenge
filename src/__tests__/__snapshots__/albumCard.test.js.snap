// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI testing - Album card 1`] = `
<div
  className="playlistCardWrapper"
>
  <img
    alt="album"
    className="albumImage"
    src="https://i.scdn.co/image/ab67616d00001e02b1494c5d42668590efcda7a2"
    width="150px"
  />
  <div
    style={
      Object {
        "paddingTop": "10px",
        "width": "190px",
      }
    }
  >
    <div
      className="title"
    >
      Y3
    </div>
    <div
      style={
        Object {
          "color": "#b3b3b3",
          "height": "40px",
          "paddingTop": "7px",
        }
      }
    >
      <div>
        <a
          className="artistName"
          href="/artist/3kHVioJpVxlazAAKQ64pC1/Yuna"
          onClick={[Function]}
        >
          Yuna
        </a>
      </div>
    </div>
  </div>
</div>
`;
